### DISCUSSION TITLE
Question regarding Features in ONLYOFFICE

### DISCUSSION THREAD

**Discussion by theezeb** [ONLYOFFICE/DocumentServer#1626] [Q&A] [answered]
Hello,
I'm trying to understand the comment feature like how does it work and connects with the text(contents),as we have selected in the Document editor and attach to it.
So i'm trying debugging in [serialize2.js](https://github.com/ONLYOFFICE/sdkjs/blob/master/word/Editor/Serialize2.js) , but i don't get it, how can i see the data of what the comment is the debugger doesn't get triggered there.

https://github.com/ONLYOFFICE/sdkjs/blob/6986086f503a1c5b85e5a195a2ded5104cfd6825/word/Editor/Serialize2.js#L16256-L16345

My Goal is to find the way it connects with content and through it, I can write my own custom comment dialog box for it tag those content with and export it in (docx,html etc)

**Did this work in previous versions of DocumentServer?**
No

**DocumentServer version:** Latest

**Operating System:** Ubuntu 20.04 LTS

**Browser version:** Version 98.0.4758.80 (Official Build) (64-bit)

**Comment by ShockwaveNN** [comment]
Moving to discussion, since it's not an issue

> how can i see the data of what the comment is the debugger doesn't get triggered there.

Are you set up the dev environment correctly? I don't think we got a guide for that and especially to adding some custom code for dialog boxes

Because is you use release version of DocumentServer - code is minimized and this method named different

Are you sure you're up to job and have enough experience, adding custom dialog box doesn't look like a trivial job

**Reply by theezeb** [reply]
> Are you set up the dev environment correctly?

Yes I have setup the DocumentServer on local with DocService , FileConverter and document-server-integration.
> Are you sure you're up to job and have enough experience
 don't think we got a guide for that

I do have enough experience to understand what i'm trying to achieve here.
Since you guys don't have guides,that's why i'm trying to debug on,how the comments work or when the files load in the editor.

>adding custom dialog box doesn't look like a trivial job

I'm not trying to added the box, i'm trying to find a way to add the comments feature with our own code which will help us tag the text content which will gives us a comment like pop-up in which our own data will  get store,

Like for example, in comments we get data like name of the user, time&date, textarea and save/cancel button
similar we want to add our own data like xbrlID,date& time,custom style(red,yellow color)on text and save/cancel buttons.

So if i were to debug the comment feature where should i start or how should begins is issue here.

**Reply by ShockwaveNN** [reply]
Are you sure your task cannot be done via plugin? Because I don't think I'll able to help you via debugging our main code, it's just above my knowledge

**Reply by theezeb** [reply]
@ShockwaveNN Yeah, we're trying to do it with custom plugins, but we're not able to get the comments once they get converted to the binary and vice versa.

it's fine if you cannot help, will try to understand it.

Can I keep this discussion open?
if I ever need to ask anything in the future.

**Reply by ShockwaveNN** [reply]
> custom plugins but we're not able get the comments once they gets converted

If you share some example of plugin we can take a look

> once they gets converted to the binary and vice versa.

But my first - maybe we simply cannot store any data to binary/docx because docx format prohibits it. 
We got a lot of troubles ourselves -  to make good comments compatible with MS Word

> Can i keep this discussion open?

Yeah, no problem

**Reply by theezeb** [reply]
> If you share some example of plugin we can take a look

Okay, will do.

![image](https://user-images.githubusercontent.com/51449693/153346738-219bae3c-56fb-468c-83ad-d112b1e01d9d.png)
I'm getting this error and in the fonts drop-down i can't select any other fonts besides `Arial`, what could be the issue here?
I tried to find whats missing on `DocumentServer` repo but didn't find any.

**Reply by ShockwaveNN** [reply]
I think the reason is that you missed this step for font data generation

https://github.com/ONLYOFFICE/build_tools#generate-fonts-data

**Reply by theezeb** [reply]
> https://github.com/ONLYOFFICE/build_tools#generate-fonts-data

tried the following steps and getting this error in terminal
`bash: server/tools/allfontsgen: No such file or directory`

**Reply by ShockwaveNN** [reply]
I think you've missed some step in complication. Which guide did you've used?

This one? https://github.com/ONLYOFFICE/build_tools#building-onlyoffice-products-source-code

Other may be oudated

**Reply by theezeb** [reply]
> Which guide did you've used?

Actually my senior followed the docker setup and he gave us the fonts folder and bin folder.
Since we don't have much ram for docker setup we used different way,but i think he as on older version DocumentServer so that's why we have issue with fonts folder.
I have written a docs setup [guide](https://docs.google.com/document/d/1hPR-CiluyOlBXz8cy4sqLaLjuDxssNfD5M-STuVHe-Q/edit?usp=sharing).

**Reply by ShockwaveNN** [reply]
> Actually my senior followed the docker setup and he gave us the fonts folder and bin folder.

So not all folders are included, I don't think it will work and may have any number of broken features because of this

Docker is not require a lot of RAM, 4 gb for build may be enough, even 2gb maybe, but always add 4-8gb swap space and build will be al-right

**Reply by theezeb** [reply]
> Docker is not require a lot of RAM, 4 gb for build may be enough, even 2gb maybe, but always add 4-8gb swap space and build will be al-right

Okay will try the docker setup, if we were to start the guide or setup where should we start? for missing files/folders.

> This one? https://github.com/ONLYOFFICE/build_tools#building-onlyoffice-products-source-code

Here? 
_sorry for the trouble._

**Reply by ShockwaveNN** [reply]
> Okay will try the docker setup, if we were to start the guide or setup where should we start? for missing files/folders.

Docker build is like one liner so I think it's easier to `compile from scratch`

https://github.com/ONLYOFFICE/build_tools#using-docker

**Comment by theezeb** [comment]


**Reply by theezeb** [reply]
nvm got it, issue was with the `path`,I saved this cmd `export PATH=/path/to/depot_tools:$PATH` in `~/.bashrc` gclient started working.

**Reply by ShockwaveNN** [reply]
BTW https://github.com/ONLYOFFICE/core/blob/1f976ae79a2593fc22ee78e9fdbb76090e83785c/DesktopEditor/doctrenderer/build_v8.txt#L22-L26

This instruction is totally outdated I think

We'll remove it

**Reply by theezeb** [reply]
Oh is there an updated instructions for it?

**Reply by ShockwaveNN** [reply]
As I've already said - [this one](https://github.com/ONLYOFFICE/build_tools#building-onlyoffice-products-source-code)

It should build all needed component and v8 also

**Reply by theezeb** [reply]
> It should build all needed component and v8 also

I think the previous source build was corrupted or broken, made another source build and now it has the v8 build too.
Thank you.

**Comment by theezeb** [comment]


**Reply by ShockwaveNN** [reply]
I'm sorry, I've never had expirience debugging our c++ code (or any c++ code), but as far as I'm aware our developers uses mostly JetBrains products, so maybe `Clion` will work, but I'm not sure

**Reply by theezeb** [reply]
Okay, no problem.

**Comment by theezeb** [comment]
I'm using the `Htmlfile2` to `docx` conversion and i noticed that there certain things missing after conversion, majority in tables
Like there are missing borders in tableCell and even width of the table is disturbed, is there anything am i missing here or it's an issue?
I have provided the html(zip) and docx
[demo3Table.docx](https://github.com/ONLYOFFICE/DocumentServer/files/8159186/demo3Table.docx)
[demo3.zip](https://github.com/ONLYOFFICE/DocumentServer/files/8159209/demo3.zip)

**Reply by ShockwaveNN** [reply]
`Htmlfile2` is not intended for direct use

Could you try to reproduce same problem with official way of converting HTML to docx via [ConvertService](https://api.onlyoffice.com/editors/conversionapi) - and if problem will be the same - you can create a new issue and we handle it

**Reply by theezeb** [reply]
> Could you try to reproduce same problem with official way of converting HTML to docx via [ConvertService](https://api.onlyoffice.com/editors/conversionapi)

okay, will let you know after trying the official conversion.

**Comment by theezeb** [comment]


**Reply by ShockwaveNN** [reply]
I'm not sure that I understand your question, since this link is part of DMS and DMS side is controlled by your side

But in your link example it's from `nodejs example` and it's source code is located [here](https://github.com/ONLYOFFICE/document-server-integration/tree/master/web/documentserver-example/nodejs)
You can debug it and get all needed information

**Reply by theezeb** [reply]
> I'm not sure that I understand your question, since this link is part of DMS and DMS side is controlled by your side
> 
> But in your link example it's from `nodejs example` and it's source code is located [here](https://github.com/ONLYOFFICE/document-server-integration/tree/master/web/documentserver-example/nodejs)
> You can debug it and get all needed information

Yeah I'm actually debugging the DMS `nodejs example` file ,but I did get the config information about the docx file.

there is an `app.js` node server  file where we the URL(editor/fileName=),but my question is that if we were to change the address for the document, we want with docID is it possible or not?

**Reply by ShockwaveNN** [reply]
I'm still not sure that I'm understand your question fully, but I think that document URL is on the DMS side (since for example with Nextcloud integration you got no file name in the URL at all) so it depends on the implementation of DMS

**Reply by theezeb** [reply]
>but I think that document URL is on the DMS side (since for example with Nextcloud integration you got no file name in the URL at all) so it depends on the implementation of DMS

Wait, we aren't using the Nextcloud integration here.


>  I'm still not sure that I'm understand your question fully, 

Sorry for lack of clarity, on localhost of ONLYOFFICE, when we open files at `localhost:3000` it sends the information regarding the DOCX file, what we're trying to do is that since it takes the fileName in the URL address, we're being thinking of adding our own documents on `localhost:3000` but with docID(not fileName), since ONLYOFFICE doesn't have`Object` for docID.

_if it's still not clear, will try to get proper info of what we're doing from my senior_

**Reply by ShockwaveNN** [reply]
> Wait, we aren't using the Nextcloud integration here.

I know - it's just an example that file name in URL is DMS responsibility

About other part - I'm sorry, I don't get it at all and I'm not very good at working with DMS, I think it's better to contact support or I leave this discussion open, maybe someone more experienced are willing to help

**Reply by theezeb** [reply]
> About other part - I'm sorry, I don't get it at all and I'm not very good at working with DMS, I think it's better to contact support or I leave this discussion open, maybe someone more experienced are willing to help

Fine, no problem at all, will try to debug and find it.
if not, will contact support for it.

**Comment by theezeb** [comment]
Hello,
I'm trying to find the documentation for Amazon S3 configuration, could please provide me the link?

**Reply by ShockwaveNN** [reply]
I don't think we got a guide for this and that using Amazon S3 as cache storage is stable feature

But you could try to change all values in [default.json](https://github.com/ONLYOFFICE/server/blob/master/Common/config/default.json#L27) to your credentials and this may work

**Reply by theezeb** [reply]
> But you could try to change all values in [default.json](https://github.com/ONLYOFFICE/server/blob/master/Common/config/default.json#L27) to your credentials and this may work

Yeah,Thanks already working on it, will ping if i got stuck after changing values.

**Reply by ShockwaveNN** [reply]
We've asked our documentation team to make some official documentation

We got some draft and it includes

1. Bucket info:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Stmt1492520443000",
            "Effect": "Allow",
            "Action": [
                "s3:ListAllMyBuckets"
            ],
            "Resource": [
                "arn:aws:s3:::*"
            ]
        },
        {
            "Sid": "Stmt1492520492000",
            "Effect": "Allow",
            "Action": [
                "s3:GetBucketLocation",
                "s3:ListBucket",
                "s3:ListBucketMultipartUploads"
            ],
            "Resource": "arn:aws:s3:::my-s3-bucketname"
        },
        {
            "Sid": "Stmt1492521245000",
            "Effect": "Allow",
            "Action": [
                "s3:AbortMultipartUpload",
                "s3:GetObject",
                "s3:ListBucket",
                "s3:ListMultipartUploadParts",
                "s3:PutObject",
                "s3:PutObjectAcl"
            ],
            "Resource": "arn:aws:s3:::my-s3-bucketname/*"
        }
    ]
 ```
  
  2. CORS:
 ```
  [
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "GET",
            "HEAD"
        ],
        "AllowedOrigins": [
            "https://documentserver_adress"
        ],
        "ExposeHeaders": []
    }
]
 ```
  
  3. "sessionabsolute": "2h" should be in default.json
  4. 
  ```
   "storage": {
                "name": "storage-s3",
                "fs": {
                        "folderPath": "",
                        "urlExpires": 900,
                        "secretString": "verysecretstring"
                },
                "region": "us-east-1",
                "endpoint": "https://s3.us-east-1.amazonaws.com/",
                "bucketName": "my-s3-bucketname",
                "storageFolderName": "files",
                "urlExpires": 604800,
                "accessKeyId": "***",
                "secretAccessKey": "***",
                "useRequestToGetUrl": true,
                "useSignedUrl": true,
                "sslEnabled": false,
                "s3ForcePathStyle": false,
                "externalHost": ""
        },
 ```

**Reply by theezeb** [reply]
Thank you for the draft much appreciated, just one question is that why is this S3 config given in (`default.json`)here, does it have any particular reason?

How to access it after giving the access and secretaccess key?

**Reply by ShockwaveNN** [reply]
> Thank you for the draft much appreciated, just one question is that why is this S3 config given in (default.json)here, does it have any particular reason?

If by this question you mean "Why not in local.json` - you can place it in local.json and it will be correct way, but default.json already has some template

> How to access it after giving the access and secretaccess key?

Not quite sure why you need that, s3 used ONLY for store temporal cache files, it's not designed to be `real storage` for user files

**Reply by theezeb** [reply]
> s3 used ONLY for store temporal cache files, 

Yeah pretty much but we have our documents saved on S3(testing) and i wanted to use my S3 config in `default.json` here so that if we upload any files from local to S3(idk if that's possible just assuming) or access files from S3 that's why i asked that question.
Since I'm just exploring the config files and everything.

**Reply by ShockwaveNN** [reply]
> our documents saved on S3(testing)

Config I've shown is not related to situation when you have real documents stored on s3, this config is used to store temporal cache files on s3

If you got real files on s3 - you should share write your own middleware which will be making links for s3 files accessible from DocumentServer host and save it back

**Reply by theezeb** [reply]
> If you got real files on s3 - you should share write your own middleware which will be making links for s3 files accessible from DocumentServer host and save it back

Okay, thank you for the information this actually might solve the problem we were thinking was the solution.

**Reply by ShockwaveNN** [reply]
I understnad what your usercase is rather complicated and involved a lot of changes in different parts of our code, so maybe consider contacting support@onlyoffice.com - they will give you more attention than I able, my team are mostly for handling bugs and supporting simple user cases

**Reply by theezeb** [reply]
> I understnad what your usercase is rather complicated and involved a lot of changes in different parts of our code, so maybe consider contacting support@onlyoffice.com - they will give you more attention than I able, my team are mostly for handling bugs and supporting simple user cases

Okay will contact them regarding my issues.
Thank you for the help,most of the changes are even harder for me too and  understand it but I'm exploring so it helps to ask it here(mostly).

Hopefully I didn't waste any of your time 😁

**Comment by theezeb** [comment]
Hello,
I did get the S3 setup running but getting CORS error after that, you have provided the CORPS in the draft but I don't know where to use it.
if you know anything about it,could you provide me the solution, i did try the online solutions but can't find the file to write it.
![image](https://user-images.githubusercontent.com/51449693/159110823-7bb4914f-6e6b-4abf-b22c-64dcf8628eda.png)

**Reply by ShockwaveNN** [reply]
I personally have never configurated it, I just copied data from pre-production help entry, but I think more details can be found [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ManageCorsUsing.html)

**Reply by theezeb** [reply]
Okay

**Comment by theezeb** [comment]
Hello,
Its been a while i have been working on the source code so i’m trying to find this icon
![image](https://user-images.githubusercontent.com/51449693/170474593-0b16fd59-0cfa-4adf-ab0b-095eb6d4bc7e.png)
I don’t want help in customising with or on it, just want to know what is the object or file name for it…
I have searched everywhere but i can’t any clue regarding that.

**Reply by ShockwaveNN** [reply]
It's rendered here as vector:
https://github.com/ONLYOFFICE/sdkjs/blob/8add64f77dc3d7dd9002ae0b6fa1c95ffc774e70/word/Drawing/DrawingDocument.js#L950

**Reply by theezeb** [reply]
> It's rendered here as vector: https://github.com/ONLYOFFICE/sdkjs/blob/8add64f77dc3d7dd9002ae0b6fa1c95ffc774e70/word/Drawing/DrawingDocument.js#L950

Thank you so much for providing the information,i thought it wasn't a part of canvas rendering at first.

**Comment by theezeb** [comment]
Hello,
I had a question regarding [Overlay](https://github.com/ONLYOFFICE/sdkjs/blob/87d3f0154f613ac750605e722c0855c458093d8d/word/Drawing/HtmlPage.js#L2995-L3115)
is there a possible way to have a static canvas on the `CPage`(after [document](https://github.com/ONLYOFFICE/sdkjs/blob/87d3f0154f613ac750605e722c0855c458093d8d/word/Drawing/DrawingDocument.js#L1908-L1979) canvas is rendered)? Like in here how the [vector](https://github.com/ONLYOFFICE/DocumentServer/discussions/1626#discussioncomment-2827015) is initialised i wanted to have my own custom vector(i made) which always shows,not like the table vector.
Since the table vector is based on mouse movement over or on table it shows and out of focus it gets removed.

But i'm not able find a proper way for it,so do you have any idea of it?

**Reply by ShockwaveNN** [reply]
Sorry, I don't think I have enough deep knowledge to answer this question at all

**Reply by theezeb** [reply]
Okay, no problem.

**Comment by theezeb** [comment]
Hello,
Do you have any idea,where will i find file for table selection when click on this [vector](https://github.com/ONLYOFFICE/DocumentServer/discussions/1626#discussioncomment-2827015) here,Is it part of sdkjs or web-apps?

**Reply by ShockwaveNN** [reply]
Hi, if by `file` you mean icon - it's part of `sdkjs` and rendered as vectored image somewhere around [here](https://github.com/ONLYOFFICE/sdkjs/blob/8add64f77dc3d7dd9002ae0b6fa1c95ffc774e70/word/Drawing/DrawingDocument.js#L992) if I'm not mistaken

**Reply by theezeb** [reply]
Yeah, I got it.

**Comment by theezeb** [comment]
Hello,
I'm using core-fonts for fonts menu but i can't find Times New Roman options in the list and even in the [repo](https://github.com/ONLYOFFICE/core-fonts/tree/d834fb1817d4264aa012528428ff3a5ba6f7f001)
Here in [demo](https://api.onlyoffice.com/editors/editor?method=docxEditor),it's showing in font options.
Can you let me know how can i get these options?

**Reply by ShockwaveNN** [reply]
We do not provide Times New Roman to the default installation because this prohibited by this font license

You can install new fonts via [this instruction](https://helpcenter.onlyoffice.com/installation/docs-community-install-fonts-linux.aspx)

**Reply by jiriks74** [reply]
@shahzebgit by Microsoft licence you cannot redistribute the fonts legally. You could probably mount the font folder from a windows install directly to /usr/share/fonts without issues as this shouldn't be redistributing. But I'm not a legal expert and I didn't read the fonts license so you should read it if you want to go that route and you really need the nicer newer fonts. You can use older Ms fonts that should be available in your repo without any legal problems (`ttf-mscorefonts-installer` if I remember it right).

**Reply by theezeb** [reply]
@jiriks74 yeah, I searched about the fonts and most answers provide `ttf-mscorefonts-installer` which we can use it.
since I'm using Source Code and for it, I'm using the build tool[ cmd line](https://github.com/ONLYOFFICE/build_tools#generate-fonts-data), I can just get the `.ttf` file and use it in my repo, as you have mentioned.

**Comment by theezeb** [comment]
Hello,
If we're using the Source Code of `ONLYOFFICE` which licensing does it fall under? since [here](https://github.com/ONLYOFFICE/DocumentServer#onlyoffice-document-server-editions) it says for 3 different editions, which doesn't say anything about the Source Code usage, could you please provide us the information regarding the licensing for it?

**Reply by ShockwaveNN** [reply]
It will be Community Edition with all the limits and the license itself for its usage is AGPL-3

**Reply by theezeb** [reply]
Thank you for the information.

**Comment by theezeb** [comment]
Hello,
I wanted to know something about the compare feature, if I have a DOCX uploaded in  my editor and then upload same DOCX but with some new changes and use compare feature over it to accept/reject the changes, but before compare when I upload the DOCX (with new changes) where does the uploaded DOCX file go? in MySQL or it gets stored in binary? what is the process in between before compare?

Do you have any resources, where can I look for this? or you have any idea?

**Reply by ShockwaveNN** [reply]
Sorry, I don't know how this works, but I don't think this file stored locally, but rather in DB, same with images I think, but I don't know details

**Reply by theezeb** [reply]
Okay, is there any documentations for it? or anything.

**Reply by ShockwaveNN** [reply]
Sorry, I don't think so, you need to dig in source code to figure it out

**Reply by theezeb** [reply]
Okay thank you, no problem, will have to check out the source code now.

**Comment by theezeb** [comment]
Hey,
We are using Amazon S3 as cache storage for our document storage etc.
 but we have certain files which are big, so the `Editor.bin`(DOCX) files takes few seconds to get downloaded and then the file gets loaded.
 
 Since you guys have https://personal.onlyoffice.com/ on their I uploaded the same file and the `Editor.bin`(DOCX) took around 1.25s to download (checking in network tab), so do you know what cache system are you guys using there?
 Do you have any other suggestion to use cache storage with Amazon S3 or something like that?

**Reply by ShockwaveNN** [reply]
Sorry, I'm not aware, and not sure that our SAAS solution is not part of NDA, I can tell you things I know about our opensource code, but SAAS is not a part of it, I think you've better to contact support@onlyoffice.com and discuss it with them

**Reply by theezeb** [reply]
Okay will contact the support regarding the SaaS part, thank you for the information.

**Comment by theezeb** [comment]
Hello,
I'm trying to implement HTTP/2 in ONLYOFFICE `server.js`, since it's in NodeJS Express, do you have any docs on how, and where should we be making changes for it to be able to work with the changes we made?

**Reply by ShockwaveNN** [reply]
Sorry, I don't think so, I think we do not provide any guides how to make any modification to our code

And personally I've not a slightest idea how to do that

**Comment by theezeb** [comment]
Hello,
We have integrated ONLY OFFICE on our application but for some reason, while using Version History we try to click on Close History
is not closing the leftMenu as it works in Standalone application of ONLY OFFICE, so is there any API call going from there which might be affecting in our application?
![as](https://user-images.githubusercontent.com/51449693/189620295-7d2b8847-7740-4ad9-9f20-807412c3060a.png)

**Reply by ShockwaveNN** [reply]
Is there any erros in javascript console?

**Reply by theezeb** [reply]
> Is there any erros in javascript console?

No there is no error,its just stuck there we have to reload instead(manually).

**Reply by ShockwaveNN** [reply]
No, I don't think I'm aware a cause of this, you should compare requests in our test example with yours

**Reply by theezeb** [reply]
Yeah it works fine there but integrating it into our application its not working,so do have any idea of the API calls from the close history button?

**Reply by ShockwaveNN** [reply]
I think all the documentation is here

https://api.onlyoffice.com/editors/history

And I see no info about close button

Seems that on pressing it - the page simply refreshed
Maybe something wrong with permissions for iframe and it do not allowing to be refreshed

**Reply by theezeb** [reply]
Thank you for the docs.

>Seems that on pressing it - the page simply refreshed
>Maybe something wrong with permissions for iframe and it do not allowing to be refreshed

Actually the issue is fixed, as it was sending a request, but our application didn't have the required API callback for it, that's why it wasn't loading.
As it was successful, that's why it was not throwing error.

**Comment by theezeb** [comment]
Hello,
What is the units used for fonts in ONLY-OFFICE like cm,pt(point),px?
Because it hard to understand since there are too many conversions,like paragraph settings has cm as a units for paragraph spacing etc.
![image](https://user-images.githubusercontent.com/51449693/190327021-ca3e4996-1c32-4bc9-8fb6-053bed510aa1.png)

**Reply by ShockwaveNN** [reply]
The font sizes in in `points` like in any other editors

> Because it hard to understand since there are too many conversions,like paragraph settings has cm as a units for paragraph spacing etc.

In Settings of DocEditor you can change units:
![image](https://user-images.githubusercontent.com/668524/190365495-d8c81280-e4ce-45ca-bc01-74724b3c5862.png)

But not all the values will be in new unit, for example Font Size will still be in points

**Comment by theezeb** [comment, marked as answer]
hello,
I'm using the Community Edition of ONLY-OFFICE and i'm using Document server integration(`nodejs`) to use the standalone version of ONLY-OFFICE i noticed that the Version History isn't working,does the version history feature works right of the bat with community edition or we have to setup some config for it? so that we can use it?

If there is please provide some docs since i looked into it and it confusing since it say some callbacks are required but are deprecated.

**Reply by ShockwaveNN** [reply]
You should be more specific by `Version History isn't working`

![image](https://user-images.githubusercontent.com/668524/190585465-7afaa134-0ebe-48b3-91c3-277360853704.png)

I've setup DocumentServer v7.1.1 CE with latest master of `nodejs` example via my  compose file:
https://github.com/ONLYOFFICE-QA/documentserver-nodejs-example-docker

And everything seems working fine

**Reply by theezeb** [reply]
> You should be more specific by `Version History isn't working`

My bad, what I mean is that I'm on ver v7.0.1 Linux setup, when I make changes in the file and check version history there are no other versions showing it only says ver.1 that's all, and I haven't changed any config, so IDK where should I start looking into.

**Reply by ShockwaveNN** [reply]
I think maybe we changed something in v7.1 format in version history, we support only latest version, so you should check v7.1.1 and see if this problem occurs

And if it occurs - you should provide some steps to reproduce

**Reply by theezeb** [reply]
> so you should check v7.1.1

Okay, will update and look into it.

**Reply by theezeb** [reply]
Could you please check if it's working fine in v7.0.1 and let me know?
since it can narrow down for me, and then I can get the proper code update with it.

**Reply by ShockwaveNN** [reply]
Version history works with 7.0.1 also
![image](https://user-images.githubusercontent.com/668524/190607076-a1330c61-00d3-40ef-8d14-f478e09a5937.png)

Which version of `nodejs` example are you using? Please specify git hash commit, since we do not versioning test example properly

**Reply by theezeb** [reply]
> Which version of `nodejs` example are you using? Please specify git hash commit, since we do not versioning test example properly

git commit hash 
05a3110c99423dafd24b6edbdda77678548c9a4d - long ver
05a3110c - short ver

**Reply by ShockwaveNN** [reply]
Same as mine, seems something else is wrong with your project

Try this as reference
https://github.com/ONLYOFFICE-QA/documentserver-nodejs-example-docker

It's setup two containers, one with DocumentServer and second one with nodejs example and this one I've used to check if everything fine

**Reply by theezeb** [reply]
> It's setup two containers, one with DocumentServer and second one with nodejs example and this one I've used to check if everything fine

Thank you will check it out and see whats the issue.

**Reply by theezeb** [reply]
I'm getting this issue after making the setup and running 4th step
```
ERROR: for documentserver-nodejs-example-docker-master_ds-test-example_1  Cannot start service ds-test-example: driver failed programming external connectivity on endpoint documentserver-nodejs-example-docker-master_ds-test-example_1 (d7aba548245be09fc1cce504db3dbfcf6a778af3631e779886bec9e2c6886dff): Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use

ERROR: for ds-test-example  Cannot start service ds-test-example: driver failed programming external connectivity on endpoint documentserver-nodejs-example-docker-master_ds-test-example_1 (d7aba548245be09fc1cce504db3dbfcf6a778af3631e779886bec9e2c6886dff): Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use
ERROR: Encountered errors while bringing up the project.
```

**Reply by theezeb** [reply]
Okay i got it working on my IP, yes its working with the docker setup but is there way i should know what is the problem?
i used the source file from GitHub and got the files from there.

**Reply by ShockwaveNN** [reply]
> listen tcp4 0.0.0.0:80: bind: address already in use

This error is literally what it's said - you got some other application on your 80 port

> there way i should know what is the problem

I'm sorry, I don't know, there may be a lot of different causes, and unless you provide some way to reproduce it - I don't think I can help you

Quite frankly you got a lot of very specific questions that make me thinking that you're using our application for commercial use, so maybe you should consider buying support option, I can help with some questions, but not all of them, and they taking a lot of my time

**Reply by ShockwaveNN** [reply]
I see in email notification that you've attached video which you've deleted for some reason

But in this video you've using Version history directly without closing file - this is not how it works at all

You should enter some text, close the file, wait for some time (around 10 seconds) reopen file - and there should be 2 version

**Reply by theezeb** [reply]
> Quite frankly you got a lot of very specific questions that make me thinking that you're using our application for commercial use, so maybe you should consider buying support option, I can help with some questions, but not all of them, and they taking a lot of my time

I'm sorry for that, didn't mean to trouble you at all, it's just asking for support I don't actually get many answers TBH, that's why I come back here to ask the question.
next time I will try to find the cause and ask here, instead of just posting it.

> But in this video you've using Version history directly without closing file - this is not how it works at all

yeah, I figured it out later what I did wrong after I reopened the file.
but for some reason, the setup I have for ONLYOFFICE the version history is not working properly.
I will have thorough investigation for it.
sorry for the inconvenience.

**Reply by ShockwaveNN** [reply]
Thanks for understanding

We do not recommend compiling our application from source if you are not a very experience user with all technologies involved (and there is a lot of different technologies involved) 

If you provide a reason why you need to use compiled version - maybe there is some workaround for this, maybe using plugins-macros os something else
